cluster-autoscaler/expander/price/preferred.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package price
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube_util "k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+
+	"github.com/golang/glog"
+)
+
+type simplePreferredNodeProvider struct {
+	nodeLister kube_util.NodeLister
+}
+
+func (spnp *simplePreferredNodeProvider) Node() (*apiv1.Node, error) {
+	nodes, err := spnp.nodeLister.List()
+	if err != nil {
+		return nil, err
+	}
+	size := len(nodes)
+	glog.Warningf("Nodes = %d", size)
+
+	mb := int64(1024 * 1024)
+	cpu := int64(1000)
+
+	// Double node size with every time the cluster size increases 3x.
+	if size <= 2 {
+		return buildNode(1*cpu, 3750*mb), nil
```

Comment:
Maybe in some later version... It doesn't matter much whether this is a machine from cloud provider or not. Overall shape/size is important.